### DISCUSSION TITLE
Are ACLs validated serverside or is it possible for a client to overwrite a synched configuration?

### DISCUSSION THREAD

**Discussion by suiluj** [gravitl/netmaker#1719] [Q&A]
Hello,

I am interested in netmaker because auf the ACLs (Access Control Lists).

At the moment I am not sure if the communication permissions are configured client-side or serverside so it would be great if someone give some clarification.

I have this question because in the normal wireguard confs there is a "Allowed IPs" entry which can be changed client side. (my setup is not using netmaker yet).

So does netmaker only change this "allowed ips" settings for every client and the netmaker client pull these settings?
This would be bad because then someone could manually change the setting client-side and still reach devices which were denied in the netmaker UI.

I hope my explanation was understandable. :)
Just wanted to be sure that netmaker restricts connections serverside independet of configurations on the client side.

**Comment by mattkasun** [comment]
Netmaker sets up peer to peer networks, disregarding gateways for the moment.  Once the peers have recieved their configuration the server is not involved in peer to peer communications, so there is no way for the server to block comms.

If as you said, someone modified their allowed ips to add a node it would work temporarily.  Netmaker sends updates to the peers periodically, so the change made to get around the acl would have to be reapplied everytime the server sends a periodic peer update

**Reply by suiluj** [reply]
ah thanks for the fast reply.
okay that's the behaviour like I imagined but not hoped for.

Because of this behaviour my understanding is that I cannot really restrict communication between devices in the network.
My goal was to only allow communication from many clients to one single device. And the many clients must not be able to see/communicate with each others.

This scenerio was described here: https://docs.netmaker.org/acls.html#scenario-1-remote-access-to-from-client-devices

I am no wireguard expert yet but could it be possible to restrict communication between clients on the netmaker server side?
As I understand in this example the allowed IPs on the serverside configuration can restrict access per peer:
https://techoverflow.net/2021/07/09/what-does-wireguard-allowedips-actually-do/

**Reply by suiluj** [reply]
I did a little research and as I understand now the Allowed IPs does not work like that.
Here is a description on how to set some interesting iptables rules in the wg post up script:
https://gist.github.com/qdm12/4e0e4f9d1a34db9cf63ebb0997827d0d#restrict-the-user-to-a-port

I guess this would be the way to restrict communication between clients.

**Reply by mattkasun** [reply]
setting iptables rules in postup/postdown suffers from the same problem.... the user can change them

**Reply by suiluj** [reply]
Well the iptable blocks are on the server

**Reply by mattkasun** [reply]
which are included in the wireguard config file sent to a node.

**Reply by suiluj** [reply]
Are you sure?
In the description of [this link is sent before](https://gist.github.com/qdm12/4e0e4f9d1a34db9cf63ebb0997827d0d#restrict-the-user-to-a-port) there are a few example on how to restrict different users to specific IPs. And as I understand these commands block the traffic on the wireguard interface in the server which forwards between the different peers. So I would think that it makes no difference what the user does on his client system the server will block unwanted traffic nervertheless.

**Reply by mattkasun** [reply]
That gist describes a different network setup where all peer to peer traffic is routed through the server.
Netmaker creates mesh networks by default where peer to peer traffic is direct and the server is not involved.
